src/azure-cli/azure/cli/command_modules/monitor/operations/general_operations.py

```diff
@@ -0,0 +1,61 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from .._client_factory import cf_monitor
+from ..util import _gen_guid
+from azure.cli.core.commands.transform import _parse_id
+from knack.log import get_logger
+from knack.util import CLIError
+from msrestazure.tools import parse_resource_id
+
+
+logger = get_logger(__name__)
+
+
+def clone_existed_settings(cmd, source_resource=None, target_resource=None):
+    metrics_alert_rules = _clone_monitor_metrics_alerts(cmd, source_resource, target_resource)
+    return metrics_alert_rules
+
+
+def _clone_monitor_metrics_alerts(cmd, source_resource, target_resource):
+    if not _is_resource_type_same(source_resource, target_resource):
+        raise CLIError('The target resource should be the same type with the source resource')
+    monitor_client = cf_monitor(cmd.cli_ctx)
+    # we can only clone the alert rules belonging to the current subscription.
+    alert_rules = list(monitor_client.metric_alerts.list_by_subscription())
+    updated_alert_rules = []
+    ErrorResponseException = cmd.get_models('ErrorResponseException', operation_group='metric_alerts')
```

Comment:
A question here: you get the client through directly import without resource type message, but here you use get_modes() here, which means you will import the model according to the api version defined in resource type.
How can you make sure the client (operations) and model are in the same api version?

Reply:
fix. init client from resource type